playground/backend/cmd/server/controller.go

```diff
@@ -53,3 +111,183 @@ func (controller *playgroundController) GetCompileOutput(ctx context.Context, in
 	compileOutput := pb.GetCompileOutputResponse{Output: "test compile output"}
 	return &compileOutput, nil
 }
+
+// runCode validates, compiles and runs code by pipelineId.
+// During each operation updates status of execution and saves it into cache.
+// In case of compilation failed saves logs to cache.
+// After success code running saves output to cache.
+func runCode(ctx context.Context, lc *fs_tool.LifeCycle, exec *executors.Executor, pipelineId uuid.UUID, timeout time.Duration) {
```

Comment:
You keep doing that :) Similarly named methods with different implementation. The first capital letter says nothing about its functionality.

Reply:
Changed.